src/IW/Core/Repo.hs

```diff
@@ -33,12 +36,33 @@ data Repo = Repo
     , repoName       :: !RepoName
     , repoDescr      :: !Text
     , repoCategories :: !(SqlArray Category)
+    , repoCabalUrl   :: !Url
     } deriving stock    (Eq, Generic, Show)
       deriving anyclass (ToJSON, FromRow, ToRow)
 
-repoUrl :: Repo -> Url
-repoUrl Repo{..} = Url
-    $ "https://github.com/"
-    <> unRepoOwner repoOwner
-    <> "/"
-    <> unRepoName repoName
+-- | Convert a value of the @GitHub.Repo@ type to a value of our own @Repo@ type.
+fromGithubRepo :: GitHub.Repo -> Repo
+fromGithubRepo githubRepo = Repo
+    { repoOwner      = owner
+    , repoName       = name
+    , repoDescr      = fromMaybe "" $ GitHub.repoDescription githubRepo
+    , repoCategories = SqlArray []
+    , repoCabalUrl   = mkRepoCabalUrl owner name $ GitHub.repoDefaultBranch githubRepo
+    }
+  where
+    owner :: RepoOwner
+    owner = RepoOwner $ GitHub.untagName $ GitHub.simpleOwnerLogin $ GitHub.repoOwner githubRepo
+
+    name :: RepoName
+    name  = RepoName $ GitHub.untagName $ GitHub.repoName githubRepo
+
+-- | This function returns a @Url@ for downloading a @Repo@'s @.cabal@ file.
+mkRepoCabalUrl :: RepoOwner -> RepoName -> Maybe Text -> Url
+mkRepoCabalUrl (RepoOwner repoOwner) (RepoName repoName) maybeBranch = Url $
+    "https://raw.githubusercontent.com/"
+    <> Text.intercalate "/"
```

Comment:
I was mostly telling about putting the hardcoded `https://...` string inside the list and merging `interclate` line with `Url $ ` line to have more consistent code, not use `<>` and have less lines

Reply:
@chshersh I missed that :man_facepalming: That's definitely better